### PR TITLE
JS-4: Implement header and footer

### PR DESCRIPTION
**Pull request by TeodorTunev**

### CODE REVIEW

File: app/views/home/home.component.ts [hunk 1]
```diff
@@ -17,8 +19,11 @@ export class HomeComponent implements OnInit {
   newJobs: Array<Job>;
   isCompanyUser: boolean;
 
-  constructor(private jobProxy: JobProxy, private authManager: AuthManager) {
-    this.isCompanyUser = authManager.isCompanyUser();
+  constructor(private jobProxy: JobProxy, private userManagerService: UserManagerService, private translationService: TranslationService) {
+    this.isCompanyUser = userManagerService.isCompanyUser();
+    this.translationService.getLanguageChangeEmiiter().subscribe(() => {
```
Comment (bvangelov) [inline app/views/home/home.component.ts]: It should be "Emitter"

File: app/app.component.html [hunk 2]
```diff
@@ -1,9 +1,93 @@
...
+          <div class="nav-menu-item clickable"><a class="nav-menu-link nav-faq" href="">{{'menu.main.faq' | translate}}</a></div>
+          <div class="nav-menu-item clickable"><a class="nav-menu-link nav-contact" href="">{{'menu.main.contact' | translate}}</a></div>
+          <div class="nav-menu-item clickable" *ngIf="user && !isCompanyUser"><a class="nav-menu-link nav-profile" routerLink="/user">{{'profile' | translate}}</a></div>
+          <div class="nav-menu-item clickable" *ngIf="!user"><a class="nav-menu-link nav-signin" routerLink="/login">{{'menu.main.login' | translate}}</a></div>
+          <div class="nav-menu-item clickable" *ngIf="user"><a class="nav-menu-link nav-signout" (click)="onLogoutButtonClick()">{{'menu.main.logout' | translate}}</a></div>
+        </div>
+      </div>
+    </div>
+    <div class="table-row nav-menu-footer">
+      <div class="table-cell user-panel" *ngIf="user"><div class="user-info-image" [ngStyle]="{'background-image': 'url(' + ((user.images && user.images[0] && user.images[0].imageUrl) || 'assets/images/placeholder-profile-image.png') + ')'}"></div>{{user.firstName}} {{user.lastName}}</div>
```
Comment (belev) [inline app/app.component.html]: move on separate line

File: app/app.component.html [hunk 3]
```diff
@@ -1,9 +1,93 @@
...
+      <div class="table-cell text-center" *ngIf="!user"><a routerLink="/user/register" class="button" >Create User</a></div>
```
Comment (belev) [inline app/app.component.html]: move on separate line and remove whitespace after "class="button" "

File: app/app.component.html [hunk 4]
```diff
@@ -1,9 +1,93 @@
...
+  </div>
+</div>
+<div class="menu-side-bar" [hidden]="!isLanguageMenuVisible" (click)="onBodyClick($event)">
+  <div class="overbody-container"></div>
+  <div class="close-button icon-ic-close" (click)="isLanguageMenuVisible = false"></div>
+  <div class="menu-container language-menu-container">
+    <div class="just-arrived-logo-container text-center">
+      <img class="language-logo-img" src="app/assets/images/logo.svg" alt="Just Arrived"/>
+    </div>
+    <div class="select-language-header-text text-center fs-xl">{{ 'menu.language.title' | translate }}</div>
```
Comment (belev) [inline app/app.component.html]: remove spaces aroung {{ }}

File: app/app.component.html [hunk 5]
```diff
@@ -1,9 +1,93 @@
...
+    <div class="select-language-container text-center">
+      <div *ngFor="let language of systemLanguages" class="language-button-container no-select clickable" (click)="onSelectLanguage(language)">
+        <div class="language-code">
+          {{language.languageCode}}
```
Comment (belev) [inline app/app.component.html]: move on one line

File: app/app.component.html [hunk 6]
```diff
@@ -1,9 +1,93 @@
...
+      </div>
+
+    </div>
+    <div class="language-information text-center">
+      <p>{{ 'menu.language.text_1' | translate }}</p>
+      <p>{{ 'menu.language.text_2' | translate }}</p>
+    </div>
+    <div class="credentials">
+      <p class="prominent">{{ 'menu.language.translation_by_probono' | translate }}:</p>
+      <img src="app/assets/images/menu_semantix_logo.png" alt="Semantix"/>
```
Comment (belev) [inline app/app.component.html]: remove "app/" from src

File: app/app.component.html [hunk 7]
```diff
@@ -1,9 +1,93 @@
...
+      <img src="app/assets/images/menu_skrivasbyra_logo.png" alt="Stockholms.Skrivbyrå"/>
```
Comment (belev) [inline app/app.component.html]: remove "app/" from src

File: app/app.component.html [hunk 8]
```diff
@@ -1,9 +1,93 @@
...
+  <div class="table-row">
+    <div class="table-cell header-logo-container">
+      <div class="table">
+        <div class="table-row">
+          <div class="table-cell clickable back-button-container" *ngIf="states.length > 1 && currentState !== '/home'" (click)="onBackButtonClick()">
+            <i class="demo-icon icon-ic-arrow-left"></i>
+          </div>
+          <div class="table-cell app-logo-img-container">
+            <a routerLink="/home">
+              <img class="app-logo-img" src="app/assets/images/logo.svg" alt="Just Arrived"/>
```
Comment (belev) [inline app/app.component.html]: remove "app/" from src

File: app/models/user.ts [hunk 9]
```diff
@@ -28,6 +28,7 @@ export class User {
     this.workExperience = jsonObject.job_experience;
     this.education = jsonObject.education;
     this.skills = jsonObject.competence_text;
+    console.log(this);
```
Comment (bvangelov) [inline app/models/user.ts]: remove debugging code

File: app/app.component.scss [hunk 10]
```diff
@@ -1,4 +1,304 @@
...
+          .nav-menu-link {
+            text-decoration: none;
+            color: $color-blue-3;
+            padding-left: 40px;
+
+            &:hover {
+              opacity: 0.7;
+            }
+
+           &:before {
```
Comment (belev) [inline app/app.component.scss]: use the icon classes from fontello

File: app/app.component.ts [hunk 11]
```diff
@@ -10,13 +13,24 @@ import {TranslationService} from "./services/translation.service";
   styleUrls: ["app.component.css"]
 })
 export class AppComponent implements OnInit {
-  constructor(private authManager: AuthManager, private router: Router, private translationService: TranslationService) {
+  states: Array<String> = new Array<String>();
```
Comment (belev) [inline app/app.component.ts]: states: string[] = [];

File: app/app.component.ts [hunk 12]
```diff
@@ -10,13 +13,24 @@ import {TranslationService} from "./services/translation.service";
...
+  currentState: string;
+  user: User;
+  isCompanyUser: boolean;
+  systemLanguages: Array<Language>;
```
Comment (belev) [inline app/app.component.ts]: Array<Language> -> Language[]

File: app/app.component.ts [hunk 13]
```diff
@@ -10,13 +13,24 @@ import {TranslationService} from "./services/translation.service";
...
+  selectedLanguage: Language;
+  isNavigationMenuVisible: boolean = false;
+  isLanguageMenuVisible: boolean = false;
+
+  constructor(private router: Router, private authManager: AuthManager, private userManagerService: UserManagerService, public translationService: TranslationService) {
```
Comment (belev) [inline app/app.component.ts]: split on multiple lines

File: app/app.component.ts [hunk 14]
```diff
@@ -31,17 +45,55 @@ export class AppComponent implements OnInit {
...
+    this.router.navigate([this.states.pop()]);
+  }
+
+  onLogoutButtonClick() {
+    this.authManager.logoutUser();
+    this.router.navigate(['/home']);
   }
 }
+
+
```
Comment (belev) [inline app/app.component.ts]: remove one line

File: app/user-manager.service.ts [hunk 15]
```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import {User} from "./models/user";
+import {LocalStorageWrapper} from "./services/local-storage-wrapper.service";
+
+@Injectable()
+export class UserManagerService {
```
Comment (bvangelov) [inline app/user-manager.service.ts]: should be only UserManager, to be consistent with other services

File: app/models/user.ts (hunk 9 above)
Comment (belev) [inline app/models/user.ts]: remove log

File: app/services/auth-manager.service.ts [hunk 16]
```diff
@@ -1,65 +1,54 @@
...
-        this.localStorageWrapper.setObject(this.storageAuthorizationData, data);
-        return this.userProxy.getUser(data['user_id']);
-      })
-      .then(response => {
-        return this.handleUserResult(response.data);
-      });
+      return this.userProxy.getUser(this.userManagerService.getUserId());
+    })
+    .then(response => {
+      return this.handleUserResult(response.data);
```
Comment (belev) [inline app/services/auth-manager.service.ts]: move on one line: response => this.handleUserResult(response.data)

File: app/services/translation.service.ts [hunk 17]
```diff
@@ -1,19 +1,48 @@
...
+import {LanguageProxy} from "./proxy/language-proxy.service";
+import {LocalStorageWrapper} from "./local-storage-wrapper.service";
+import {orderBy} from "lodash";
+import {Language} from "../models/language/language";
 
 @Injectable()
 export class TranslationService {
-  private translateService: TranslateService;
+  private storageSelectedLanguageKey: string = 'selectedLanguage';
+  private languages: Array<Language>;
```
Comment (belev) [inline app/services/translation.service.ts]: Language[]

File: app/services/user-proxy.service.ts [hunk 18]
```diff
@@ -13,6 +13,13 @@ export class UserProxy {
       return this.apiCall.get('users/' + userId, includes);
     }
 
+    getUserSession(email, password) {
+      return this.apiCall.post('users/sessions', {
+        "email-or-phone": email,
```
Comment (belev) [inline app/services/user-proxy.service.ts]: use underscores

File: app/views/login/login.component.ts [hunk 19]
```diff
@@ -0,0 +1,21 @@
...
+})
+export class LoginComponent {
+  email: string;
+  password: string;
+
+  constructor(private authManager: AuthManager, private router: Router) { }
+
+  onLoginButtonClick() {
+    this.authManager.logUser(this.email, this.password).then(result => {
+      this.router.navigate(['/home']);
```
Comment (belev) [inline app/views/login/login.component.ts]: move on one line

**Review by bvangelov — APPROVED** [review]

**Pull request merged**